libqtile/images.py

```diff
@@ -0,0 +1,205 @@
+from __future__ import division
+import cairocffi
+import cairocffi.pixbuf
+import io
+import os
+from collections import namedtuple as _namedtuple
+
+class LoadingError(Exception):
+    pass
+
+_SurfaceInfo = _namedtuple('_SurfaceInfo', ('surface', 'file_type'))
+def get_cairo_surface(bytes_img):
+    try:
+        surf = cairocffi.ImageSurface.create_from_png(io.BytesIO(bytes_img))
+        return _SurfaceInfo(surf, 'png')
+    except (MemoryError, OSError):
+        pass
+    try:
+        surf, fmt = cairocffi.pixbuf.decode_to_image_surface(bytes_img)
+        return _SurfaceInfo(surf, fmt)
+    except cairocffi.pixbuf.ImageLoadingError:
+        pass
+    raise LoadingError("Couldn't load image!")
+
+def get_cairo_pattern(surface, width, height, theta=0.0):
+    """Return a SurfacePattern from an ImageSurface.
+
+    if width and height are not None scale the pattern
+    to be size width and height.
+
+    theta is in degrees ccw
+    """
+    EPSILON = 1.0e-6
+    from math import pi
+
+    if surface is None:
```

Comment:
Mmm not sure I'll check. I took this function from another branch where it might've been None.